### PR TITLE
Tolerence of 10^-14 for even spacing too strict

### PR DESCRIPTION
**Pull request by ashjbarnes**
The assertion that the grid spacing must not vary by more than 10^-14 fails for high enough resolution runs. To replicate, set `resolution = 0.0125`, with an `xextent` and `yextent` of 30 degrees or so. What happens is that because the memory is symmetric, we need to have an odd number of longitude points so that it starts and finishes with a q point. I guess that dividing by an odd number results in some numerical error, which for high enough resolution exceeds 10^-14.

I suggest changing this tolerance to 10^-10

### CODE REVIEW

**Comment by navidcy** [comment]
For the case you suggested above I get:

```python
>>> λ = np.linspace(0, 30, 30*8+1)
>>> dλ = λ[1] - λ[0]
>>> np.max(np.diff(λ) - dλ) < 1e-14
True
```

So perhaps the culprit is something else?

**Comment by navidcy** [comment]
> I guess that dividing by an odd number...

where is the division with the number of longitude points happen?

**Comment by ashjbarnes** [comment]
> > I guess that dividing by an odd number...
> 
> where is the division with the number of longitude points happen?

```
import numpy as np

yextent = [-56,-26]
xextent = [142,180]
resolution = 0.0125

nx = int((xextent[1] - xextent[0]) / (resolution / 2))
print(nx)
if nx % 2 != 1:
    nx += 1
print(nx)
λ = np.linspace(
    xextent[0], xextent[1], nx
)  # longitudes in degrees
dλ = λ[1] - λ[0]

print(np.max(np.diff(λ) - dλ))
assert (
    np.max(np.diff(λ) - dλ) < 1e-14
), "provided array of longitudes must be uniformly spaced"
```
This test fails on numpy `1.21.5`, python `3.10.12`

File: regional_mom6/regional_mom6.py [hunk 1]
```diff
@@ -440,7 +440,7 @@ def rectangular_hgrid(λ, φ):
     dλ = λ[1] - λ[0]
 
     assert (
-        np.max(np.diff(λ) - dλ) < 1e-14
+        np.max(np.diff(λ) - dλ) < 1e-10
```
Comment (navidcy) [inline regional_mom6/regional_mom6.py:443]: The function `numpy.allclose` takes into account numerical tolerances better.

```suggestion
        np.allclose(np.diff(λ), dλ * np.ones(np.size(λ)-1))
```

Indeed the snippet you posted was failing for me also. The code I suggest above works for even down to `resolution = 0.000001`...!

**Review by navidcy — CHANGES_REQUESTED** [review]
I suggested an alternative that doesn't hardcode any tolerance value :)

**Comment by navidcy** [comment]
We should also update

https://github.com/COSIMA/regional-mom6/blob/aa624c951df49cc8d95e7793ec4ffff98ba93b2f/regional_mom6/regional_mom6.py#L209-L211

as well.

**Comment by navidcy** [comment]
perhaps we can resolve #115 here while we are at it?

**Review by navidcy — APPROVED** [review]
lgtm!

**Pull request merged**